lib/pbench/server/api/resources/upload_api.py

```diff
@@ -143,33 +144,36 @@ def put(self, filename: str):
             bytes_received = 0
 
             self.logger.info(
-                "PUT uploading {}:{} for user {} to {}",
+                "PUT uploading {}:{} for user {}|{} to {}",
                 controller,
                 filename,
+                user_id,
                 username,
                 tar_full_path,
             )
 
             # Create a tracking dataset object; it'll begin in UPLOADING state
             try:
                 dataset = Dataset(
-                    owner=username,
+                    owner_id=user_id,
                     name=Dataset.stem(tar_full_path),
                     resource_id=md5sum,
                 )
                 dataset.add()
             except DatasetDuplicate:
                 dataset_name = Dataset.stem(tar_full_path)
                 self.logger.info(
-                    "Dataset already exists, user = {}, file = {!a}",
+                    "Dataset already exists, user = {}|{}, file = {!a}",
```

Comment:
I used pipes to separate the fields of a dataset, but that was mostly so it'd be visually obvious it's a dataset dump. While I'm not attached to the `User.__str__` of `f"User, id: {self.id}, username: {self.username}"`, I'm not crazy about the pipes here.

Reply:
Yeah but the pipe in this case sort of lets us keep the `user_id` and `username` as user identifiers together, doesn't hurt to have a username with user_id in our logs especially when the user is uploading the data.

Reply:
Since `Dataset.__str__` separates user and dataset info with a pipe, you've made this message resemble that one and that confuses the context. (And, yeah, the fact that I named "drb"'s dataset "drb", though it was convenient at the time, doesn't really help.) Anyway, I'd vastly prefer to avoid using the unqualified `|` separator for anything else that might appear in dataset contexts, and use a comma or an `=` or most anything else to separate `user_id` from `user_name` in contexts like this where it might easily be confusing.